validation/provisioning/k3s/hardened_test.go

```diff
@@ -0,0 +1,138 @@
+//go:build validation || recurring
+
+package k3s
+
+import (
+	"os"
+	"testing"
+
+	"github.com/rancher/shepherd/clients/ec2"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	extensionscluster "github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/config/operations"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/rancher/tests/actions/charts"
+	"github.com/rancher/tests/actions/clusters"
+	"github.com/rancher/tests/actions/config/defaults"
+	"github.com/rancher/tests/actions/projects"
+	"github.com/rancher/tests/actions/provisioning"
+	"github.com/rancher/tests/actions/provisioninginput"
+	"github.com/rancher/tests/actions/qase"
+	"github.com/rancher/tests/actions/reports"
+	cis "github.com/rancher/tests/validation/provisioning/resources/cisbenchmark"
+	standard "github.com/rancher/tests/validation/provisioning/resources/standarduser"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type hardenedTest struct {
+	client              *rancher.Client
+	session             *session.Session
+	standardUserClient  *rancher.Client
+	cattleConfig        map[string]any
+	project             *management.Project
+	chartInstallOptions *charts.InstallOptions
+}
+
+func hardenedSetup(t *testing.T) hardenedTest {
+	var k hardenedTest
+	testSession := session.NewSession()
+	k.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	assert.NoError(t, err)
+
+	k.client = client
+
+	k.cattleConfig = config.LoadConfigFromFile(os.Getenv(config.ConfigEnvironmentKey))
+
+	k.cattleConfig, err = defaults.LoadPackageDefaults(k.cattleConfig, "")
+	assert.NoError(t, err)
+
+	k.cattleConfig, err = defaults.SetK8sDefault(client, "k3s", k.cattleConfig)
+	assert.NoError(t, err)
+
+	k.standardUserClient, err = standard.CreateStandardUser(k.client)
+	assert.NoError(t, err)
+
+	return k
+}
+
+func TestHardened(t *testing.T) {
+	t.Parallel()
+	k := hardenedSetup(t)
+	nodeRolesStandard := []provisioninginput.MachinePools{provisioninginput.EtcdMachinePool, provisioninginput.ControlPlaneMachinePool, provisioninginput.WorkerMachinePool}
+
+	nodeRolesStandard[0].MachinePoolConfig.Quantity = 3
+	nodeRolesStandard[1].MachinePoolConfig.Quantity = 2
+	nodeRolesStandard[2].MachinePoolConfig.Quantity = 3
+
+	tests := []struct {
+		name            string
+		client          *rancher.Client
+		machinePools    []provisioninginput.MachinePools
+		scanProfileName string
+	}{
+		{"K3S_CIS_1.9_Profile|3_etcd|2_cp|3_worker", k.standardUserClient, nodeRolesStandard, "k3s-cis-1.9-profile"},
```

Comment:
for CIS profile, I doesn't that change somewhat often? If so, wouldn't it make sense to make this less hardcoded in some way, i.e. a regex on the profile version or something?